.github/workflows/publish.yaml

```diff
@@ -0,0 +1,56 @@
+name: Build & Publish
+on: [workflow_dispatch]
+
+jobs:
+  build:
+    name: release
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: Use Node.js
+      uses: actions/setup-node@v3
+      with:
+        node-version: '16.20.1'
+    # Run install dependencies
+    - name: Install dependencies
+      run: npm install
+    # Run tests
+    - name: Build 
+      run: npm run compile
+    - name: Get current package version
+      id: package_version
+      uses: martinbeentjes/npm-get-version-action@v1.1.0
+    - name: Check version is mentioned in Changelog
+      id: changelog_reader
+      uses: mindsers/changelog-reader-action@v2
+      with:
+        validation_depth: 10
+        version: ${{ steps.package_version.outputs.current-version }}
+        path: 'CHANGELOG.md'
+    - name: Create a Release
+      id: create_release
+      uses: actions/create-release@v1
+      env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} 
+      with:
+        tag_name : ${{ steps.package_version.outputs.current-version}}
+        release_name: ${{ steps.package_version.outputs.current-version}}
+        body: Publish ${{ steps.changelog_reader.outputs.changes }}
+    - name: Create vsix and publish to marketplace
+      id: create_vsix
+      uses: HaaLeo/publish-vscode-extension@dfe4f6ad46624424fe24cb5bca79839183399045 #v1
```

Comment:
Please consider calling directly VSCE to publish the extension, so that we don't share our VS_MARKETPLACE_TOKEN with a third-party tool that could be compromised in the future. Or maybe we can replicate what this other Github Action is doing so that we don't reinvent the wheel?

Reply:
@daniv-msft Thanks for the suggestion. I looked into the code for this GH action extension -
1. it is tied to a specific commit so even in the future if this action has changed which might be vulnerable, we won't use it.
2. this specific commit VSIX publishing part doesn't log or steal the PAT information anywhere - [here](https://github.com/HaaLeo/publish-vscode-extension/blob/dfe4f6ad46624424fe24cb5bca79839183399045/src/publish.ts#L11) is the commit for it.
3. we will consider writing our own VSIX publishing but since this action already handles it for us , we are reusing it. 

Please let us know for further concerns.